pkg/syncer/metadatasyncer.go

```diff
@@ -2443,19 +2443,33 @@ func csiPVDeleted(ctx context.Context, pv *v1.PersistentVolume, metadataSyncer *
 
 		// If it is a multi VC setup, then skip this volume as we do not support file share volumes
 		// on a multi VC deployment
-		if isMultiVCenterFssEnabled && len(metadataSyncer.configInfo.Cfg.VirtualCenter) > 1 {
+		if isMultiVCenterFssEnabled && len(metadataSyncer.configInfo.Cfg.VirtualCenter) > 1 &&
+			!metadataSyncer.coCommonInterface.IsFSSEnabled(ctx,
+				common.TopologyAwareFileVolume) {
 			log.Debugf("PVDeleted: %q is a vSphere volume claim in namespace %q."+
 				"File share volumes are not supported in a multi VC setup."+
 				"Skipping deletion of PV metadata.", pv.Name, pv.Namespace)
 			return
 		}
-
-		// Setting volumeHandle as empty as there is only 1 VC so volumeID does not matter.
-		vcHost, cnsVolumeMgr, err := getVcHostAndVolumeManagerForVolumeID(ctx, metadataSyncer, "")
-		if err != nil {
-			log.Errorf("PVDeleted: Failed to get VC host and volume manager for single VC setup. "+
-				"Error occoured: %+v", err)
-			return
+		var vcHost string
+		var cnsVolumeMgr volumes.Manager
+		var err error
+		if metadataSyncer.coCommonInterface.IsFSSEnabled(ctx,
+			common.TopologyAwareFileVolume) {
```

Comment:
do we need this code for single vCenter with TopologyAwareFileVolume FSS enabled?

Reply:
This code path was always there for single VC/multi VC